nexus/inventory/src/builder.rs

```diff
@@ -387,6 +402,105 @@ impl CollectionBuilder {
     pub fn found_error(&mut self, error: InventoryError) {
         self.errors.push(error);
     }
+
+    /// Record information about a sled that's part of the control plane
+    pub fn found_sled_inventory(
+        &mut self,
+        source: &str,
+        inventory: sled_agent_client::types::Inventory,
+    ) -> Result<(), anyhow::Error> {
+        let sled_id = inventory.sled_id;
+
+        // Normalize the baseboard id, if any.
+        use sled_agent_client::types::Baseboard;
+        let baseboard_id = match inventory.baseboard {
+            Baseboard::Pc { .. } => None,
+            Baseboard::Gimlet { identifier, model, revision: _ } => {
+                Some(Self::normalize_item(
+                    &mut self.baseboards,
+                    BaseboardId {
+                        serial_number: identifier,
+                        part_number: model,
+                    },
+                ))
+            }
+            Baseboard::Unknown => {
+                self.found_error(InventoryError::from(anyhow!(
+                    "sled {:?}: reported unknown baseboard",
+                    sled_id
+                )));
+                None
+            }
+        };
+
+        // Socket addresses come through the OpenAPI spec as strings, which
```

Comment:
Not going to complain about json and openapi, not going to complain about json and openapi... strings.... gah...

Reply:
Yeah...it's on my list to see if we can't do better here.  It seems like it should be possible to preserve the SocketAddr types across the wire like we do for IP addresses.

Reply:
It's really not a big deal. I just had to complain about something haha.